doc/dev/index.rst

```diff
@@ -165,6 +165,41 @@ Building from source
 
 See instructions at :doc:`/install/build-ceph`.
 
+Using ccache to speed up local builds
+-------------------------------------
+
+Rebuilds of the ceph source tree can benefit significantly from use of `ccache`_.
+Many a times while switching branches and such, one might see build failures for
+certain older branches mostly due to older build artifacts. These rebuilds can
+significantly benefit the use of ccache. For a full clean source tree, one could
+do ::
+
+  $ make clean
+
+  # note the following will nuke everything in the source tree that
+  # isn't tracked by git, so make sure to backup any log files /conf options
+
+  $ git clean -fdx; git submodule foreach git clean -fdx
+
+ccache is available as a package in most distros. It is recommended to set
+symlinks to ccache to your compiler for speeding up all builds. This can be done
+by symlinking ccache to the compiler, assuming ccache is installed and the
+binary is in the standard location  ``/usr/bin/ccache`` ::
+
+  $ ln -s /usr/bin/ccache /usr/local/bin/gcc
```

Comment:
to build ceph with ccache, one can:
```
cmake -DWITH_CCACHE=ON ..
```

to enable the cmake build without pointing gcc/g++/cc/c++ to ccache.